coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1511,6 +1517,36 @@ func (s *server) CompleteJob(ctx context.Context, completed *proto.CompletedJob)
 	return &proto.Empty{}, nil
 }
 
+func (s *server) notifyWorkspaceDeleted(ctx context.Context, workspace database.Workspace, build database.WorkspaceBuild) {
+	var reason string
+	if build.Reason.Valid() {
+		switch build.Reason {
+		case database.BuildReasonInitiator:
+			if build.InitiatorID == workspace.OwnerID {
+				// Deletions initiated by self should not notify.
+				return
+			}
+
+			reason = fmt.Sprintf("initiated by _%s_", build.InitiatorByUsername)
+		case database.BuildReasonAutodelete:
+			reason = "autodeleted due to dormancy"
+		default:
+			reason = string(build.Reason)
+		}
+	}
```

Comment:
If the build reason is non-empty, but invalid (e.g. a value not understood by this code, possibly due to versioning issues), I think we should use it anyway.

Any time it's invalid we should drop a warning log, though.